packages/vite/src/node/utils.ts

```diff
@@ -512,3 +512,19 @@ export function resolveHostname(
 
   return { host, name }
 }
+
+export function isTargetNode(target: string | false | string[]): boolean {
+  if (!target) {
+    return false
+  }
+
+  if (typeof target === 'string') {
+    return target.includes('node')
+  }
+
+  if (Array.isArray(target)) {
+    return Boolean(target.filter((f) => f.includes('node')).length)
```

Comment:
**suggestion:** I think we can use [`some`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here instead

Reply:
@Shinigami92 makes sense. Updating PR with this